stacker_blueprints/firehose.py

```diff
@@ -140,123 +133,126 @@ def kms_key_policy(key_use_arns, key_admin_arns):
+        statements.append(
+            Statement(
+                Sid="Allow use of the key",
+                Effect=Allow,
+                Principal=AWSPrincipal(key_use_arns),
+                Action=[
+                    awacs.kms.Encrypt,
+                    awacs.kms.Decrypt,
+                    awacs.kms.ReEncrypt,
+                    awacs.kms.GenerateDataKey,
+                    awacs.kms.GenerateDataKeyWithoutPlaintext,
+                    awacs.kms.DescribeKey,
+                ],
+                Resource=["*"]
+            )
         )
-    )
-    statements.append(
-        Statement(
-            Sid="Allow attachment of persistent resources",
-            Effect=Allow,
-            Principal=AWSPrincipal(key_use_arns),
-            Action=[
-                awacs.kms.CreateGrant,
-                awacs.kms.ListGrants,
-                awacs.kms.RevokeGrant,
-            ],
-            Resource=["*"],
-            Condition=Condition(Bool("kms:GrantIsForAWSResource", True))
+
+        statements.append(
+            Statement(
+                Sid="Allow attachment of persistent resources",
+                Effect=Allow,
+                Principal=AWSPrincipal(key_use_arns),
+                Action=[
+                    awacs.kms.CreateGrant,
+                    awacs.kms.ListGrants,
+                    awacs.kms.RevokeGrant,
+                ],
+                Resource=["*"],
+                Condition=Condition(Bool("kms:GrantIsForAWSResource", True))
+            )
         )
-    )
-    statements.append(
-        Statement(
-            Sid="Allow access for Key Administrators",
-            Effect=Allow,
-            Principal=AWSPrincipal(key_admin_arns),
-            Action=[
-                Action("kms", "Create*"),
-                Action("kms", "Describe*"),
-                Action("kms", "Enable*"),
-                Action("kms", "List*"),
-                Action("kms", "Put*"),
-                Action("kms", "Update*"),
-                Action("kms", "Revoke*"),
-                Action("kms", "Disable*"),
-                Action("kms", "Get*"),
-                Action("kms", "Delete*"),
-                Action("kms", "ScheduleKeyDeletion"),
-                Action("kms", "CancelKeyDeletion"),
-            ],
-            Resource=["*"],
+
+    if key_admin_arns:
+        statements.append(
+            Statement(
+                Sid="Allow access for Key Administrators",
+                Effect=Allow,
+                Principal=AWSPrincipal(key_admin_arns),
+                Action=[
+                    Action("kms", "Create*"),
+                    Action("kms", "Describe*"),
+                    Action("kms", "Enable*"),
+                    Action("kms", "List*"),
+                    Action("kms", "Put*"),
+                    Action("kms", "Update*"),
+                    Action("kms", "Revoke*"),
+                    Action("kms", "Disable*"),
+                    Action("kms", "Get*"),
```

Comment:
It's actually not - since this is a KeyPolicy, it's attached directly to the key itself (not like most IAM policies) so it only applies to this specific key, and honestly, we should probably just do `Action("kms", "*")` probably at this point.